File: src/main/java/burst/pool/miners/PoolFeeRecipient.java
package burst.pool.miners;

import burst.kit.entity.BurstAddress;
import burst.kit.entity.BurstValue;
import burst.pool.storage.config.PropertyService;
import burst.pool.storage.config.Props;
import burst.pool.storage.persistent.MinerStore;

public class PoolFeeRecipient implements Payable {
    private final PropertyService propertyService;
    private final MinerStore.FeeRecipientStore store;

    public PoolFeeRecipient(PropertyService propertyService, MinerStore.FeeRecipientStore store) {
        this.propertyService = propertyService;
        this.store = store;
    }

    @Override
    public void increasePending(BurstValue delta) {
        store.setPendingBalance(store.getPendingBalance()
                .add(delta));
    }

    @Override
    public void decreasePending(BurstValue delta) {
        store.setPendingBalance(store.getPendingBalance().subtract(delta));
    }

    @Override
    public BurstValue getMinimumPayout() {
        return BurstValue.fromBurst(propertyService.getFloat(Props.defaultMinimumPayout));
    }

    @Override
    public BurstValue takeShare(BurstValue availableReward) {
        return BurstValue.fromBurst(0);
    }

    @Override
    public BurstValue getPending() {
        return store.getPendingBalance();
    }

    @Override
    public BurstAddress getAddress() {
        return propertyService.getBurstAddress(Props.feeRecipient);
    }
}

File: src/main/kotlin/burst/pool/miners/PoolFeeRecipient.kt
package burst.pool.miners

import burst.kit.entity.BurstAddress
import burst.kit.entity.BurstValue
import burst.pool.storage.config.PropertyService
import burst.pool.storage.config.Props
import burst.pool.storage.persistent.MinerStore.FeeRecipientStore

class PoolFeeRecipient(private val propertyService: PropertyService, private val store: FeeRecipientStore) : Payable {
    override fun increasePending(delta: BurstValue) {
        store.pendingBalance = store.pendingBalance
                .add(delta)
    }

    override fun decreasePending(delta: BurstValue) {
        store.pendingBalance = store.pendingBalance.subtract(delta)
    }

    override val minimumPayout: BurstValue
        get() = BurstValue.fromBurst(propertyService.get(Props.defaultMinimumPayout).toDouble())

    override fun takeShare(availableReward: BurstValue): BurstValue {
        return BurstValue.fromBurst(0.0)
    }

    override val pending: BurstValue
        get() = store.pendingBalance

    override val address: BurstAddress
        get() = propertyService.get(Props.feeRecipient)!!
}
